specs/interop/token-bridging.md

```diff
@@ -2,96 +2,296 @@
-An issue with bridging is that if the security model is not standardized,
-then the bridged assets are not fungible with each other depending on
-which bridge solution was used to facilitate the transfer between
-domains.
+## Overview
 
-## Interface
+Without a standardized security model, bridged assets may not be fungible with each other. The `SuperchainERC20` unifies ERC20 token bridging to make it fungible across the Superchain. It builds on top of the messaging protocol, as the most trust minimized bridging solution.
 
-The `ISuperchainERC20` interface creates a standard way to send tokens
-between domains. In contrast to [xERC20][xerc20] where the user calls
-a bridge that has `mint` and `burn` privileges on the cross chain enabled
-tokens, the user instead calls methods on the token contract directly.
+Unlike other standards, such as [xERC20](https://www.xerc20.com/), where users interact with a bridge possessing mint and burn privileges on cross-chain enabled tokens, this approach allows users to call methods directly on the token contract.
 
-[xerc20]: https://www.xerc20.com/
+## Implementation
 
-```solidity
-interface ISuperchainERC20 {
-  function bridgeERC20(uint256 amount, uint256 chainId) external;
-  function bridgeERC20To(address to, uint256 amount, uint256 chainId) external;
-  function finalizeBridgeERC20(address to, uint256 amount) external;
-}
-```
+The standard will build on top of ERC20 and include the following three external functions:
 
-## Implementation
+- `sendERC20(uint256 amount, uint256 chainId)(bool)`
+- `sendERC20To(address to, uint256 amount, uint256 chainId)(bool)`
+- `sendERC20To(address to, uint256 amount, uint256 chainId, bytes data)(bool)`
+- `finalizeSendERC20(address to, uint256 amount)`
+
+`sendERC20(uint256 amount, uint256 chainId)` and `sendERC20To(address to, uint256 amount, uint256 chainId)` will burn `amount` tokens and initialize a message to the `L2ToL2CrossChainMessenger` to mint the `amount` in the target address at `chainId`. `sendERC20To(address to, uint256 amount, uint256 chainId, bytes data)` will do the same, but also include a message.
+
+`finalizeSendERC20(address to, uint256 amount)` will process incoming messages from the `L2ToL2CrossChainMessenger` initiated from the same token on a different `chainId` and mint `amount` to the `to` address.
 
 An example implementation that depends on deterministic deployments across chains
 for security is provided. This construction builds on top of the [L2ToL2CrossDomainMessenger][l2-to-l2]
 for both replay protection and domain binding.
 
 [l2-to-l2]: ./predeploys.md#l2tol2crossdomainmessenger
 
+[TODO] should we add a sendERC20To specific event?
+
 ```solidity
-contract OptimismSuperchainERC20 is ERC20, ISuperchainERC20 {
-  constructor(string memory _name, string memory _symbol, uint256 _decimals) ERC20(_name, _symbol, _decimals) {}
+function sendERC20(uint256 _amount, uint256 _chainId) external returns (bool) {
```

Comment:
I would suggest diagrams over code for design docs

Reply:
added a mermaid flow: https://github.com/ethereum-optimism/specs/blob/25c6d31850a1919ab4fe819c7634a09572ad8add/specs/interop/token-bridging.md#diagram